### ISSUE TITLE
DTSU666-20 (SUNGROW)

### ISSUE THREAD

**Issue by RafAustralia** [lmatula/ha_chint_pm#23]
Hey - hello - very fresh here and kind of beginner.

My new system is being proposed to have chint meter dtsu666-20 attached directly to the logger, unlike now where it is attached to the inverter directly.

I have uploaded your integration to my HA and wanted to ack couple question please:

1. would it work for Sungrow chint DTSU666-20 first of all?
2. would it work for DTSU666-20 via tcp ip, or ipaddress of the logger?
3. is there a way of testing if the chint would be able to be contacted while it is still connected to inverter?
4. would I be able to somehow see what registers get pulled using this integration?

I need meter reads for my automation, and when connected to the logger, the logger hides chint data, and these registers need to be accessed as a separate slave 5 modbus... no longer as slave1 which is the inverter. (so my current meter values would disappear)

I have been given float values for each ang every register I need - just not sure how to configure these... and came across this integration hoping it would simplify the process.

Forgive me if these questions sound silly - but I am trying to find out if I can make new config work.

any comments would be appreciated.

![IMG_9103](https://github.com/user-attachments/assets/3bed7b5c-4cdd-4396-9e01-830b7eb2719d)

**Comment by lmatula** [comment]
Hi, 
Long story short, I think will not work. 
DTSU666 has existed many variant, different modbus register addresses. 
And logger maybe remap registers, or hide total for tcp/ip.

**Comment by RafAustralia** [comment]
Heeey!
Thank u so much for saying hi.
Sungrow r&d team from china has given me hand written registers float numbers and yesterday I have started converting these to a modbus style register.
Work in progress!.

Im not a coder but Im sort of hopeful it might work.
Also dtsu666-20 has rj45 port which is using an ethernet type rs485 cable. I think although this port is not a dedicated Lan it should be possible to plug it to a switch and maybe get an ip assigned directly.

Example of float values (right) converted to modbus yaml.

[image0.png]

Again, Im not a coder but hopefully I can get somewhere.  I will try.

Very grateful for ur reply. Thank u.

Raf

Sent from Nam Concept’s iPhone

On 30 Jul 2024, at 12:36 AM, lmatula ***@***.***> wrote:

﻿

Hi,
Long story short, I think will not work.
DTSU666 has existed many variant, different modbus register addresses.
And logger maybe remap registers, or hide total for tcp/ip.
